src/framework/midi/internal/platform/lin/alsamidiinport.cpp

```diff
@@ -53,12 +72,12 @@ AlsaMidiInPort::~AlsaMidiInPort()
     }
 }
 
-std::vector<MidiDevice> AlsaMidiInPort::devices() const
+MidiDeviceList AlsaMidiInPort::devices() const
 {
```

Comment:
We need to add a mutex, now this method is called from two threads. Or you need to make sure that this method is not called from two threads

Reply:
Also for other platforms